src/Nox.Lib/Factories/Types/NoxTypeYearFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Nox.Solution;
+using Nox.Types;
+
+namespace Nox.Factories.Types;
+
+public class NoxTypeYearFactory : NoxTypeFactoryBase<Year>
+{
+    public NoxTypeYearFactory(NoxSolution solution) : base(solution)
+    {
+    }
+
+    public override Year? CreateNoxType(NoxSimpleTypeDefinition simpleTypeDefinition, dynamic? value)
+        => value is null ? null :Year.From(value);
```

Comment:
We should be taking options from simpleTypeDefinition, as it was done here https://github.com/NoxOrg/Nox.Generator/blob/e3a689aa33bcce28a07dac21b510e563bfbaf4e8/src/Nox.Lib/Factories/Types/NoxTypeAreaFactory.cs